apps/wistia-videos/src/functions/getVideos.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  Project,
+  ProjectReduced,
+  WistiaError,
+  WistiaVideo,
+  WistiaProject,
+} from '../components/helpers/types';
+
+export const fetchProjects = async (bearerToken: string): Promise<WistiaProject[]> => {
+  const projects = await (
+    await fetch(`https://api.wistia.com/v1/projects.json`, {
+      headers: {
+        Authorization: `Bearer ${bearerToken}`,
+      },
+    })
+  ).json();
+  if (projects.error) {
+    throw new WistiaError({ message: projects.error, code: projects.code });
+  } else {
+    const reducedProjects = projects.map(({ id, hashedId, name }: ProjectReduced) => {
+      return { id, hashedId, name };
+    });
+
+    return reducedProjects;
+  }
+};
+
+const getTotalPages = (count: number) => {
+  return Math.ceil(count / 100);
+};
+
+export const fetchVideos = async (
+  projectIds: string[],
+  bearerToken: string
+): Promise<WistiaVideo[]> => {
+  const mappedProjects = await Promise.all(
+    projectIds.map(async (id: string) => {
+      const project = await (
+        await fetch(`https://api.wistia.com/v1/projects/${id}.json`, {
+          headers: {
+            Authorization: `Bearer ${bearerToken}`,
+          },
+        })
+      ).json();
+      let videos = project.medias;
+      if (project.mediaCount > 100) {
+        // start at 2 since the first page is already gotten
+        for (let i = 2; i <= getTotalPages(project.mediaCount); i++) {
+          const additionalProject = await (
+            await fetch(`https://api.wistia.com/v1/projects/${id}.json?page=${i}`, {
```

Comment:
I think your `promise.All` call is the right move here with regards to your `projectIds` array. But for these loops when there's > 100 media count, is there no way to just get all of them in one api call and then you can chop them up yourself?

Does calling `https://api.wistia.com/v1/projects/${id}.json` only give you the first page?
If so, why not follow the same pattern and use `Promise.All` here as well?